js/data/bucket/symbol_bucket.js

```diff
@@ -62,11 +62,11 @@ var layoutVertexArrayType = new Bucket.VertexArrayType([{
 }, {
     name: 'a_data1',
     components: 4,
-    type: 'Uint8'
+    type: 'Uint16'
 }, {
     name: 'a_data2',
     components: 2,
-    type: 'Uint8'
+    type: 'Uint16'
```

Comment:
The size of this array has a huge effect on GL JS performance. 

There are a couple of members of `data1` that don't need their size increased and no members of `data2` that need their size increased. 

I encourage you to take a closer look at the [code that defines the array structure](https://github.com/mapbox/mapbox-gl-js/blob/upgrade-geojson-vt/js/data/bucket/symbol_bucket.js#L54-L55) and the [code that populates the array](https://github.com/mapbox/mapbox-gl-js/blob/upgrade-geojson-vt/js/data/bucket/symbol_bucket.js#L74-L90). Is there any way we can modify this code to make the array smaller? Make the code more readable? Think big 😄 .